### PR TITLE
Update .gitignore to new `dart_tool` pub cache

### PR DESCRIPTION
**Pull request by bcko**
dart-lang/sdk#32030

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@ packages
 pubspec.lock
 /.pub
 .packages
+.dart_tool/
```
Comment (kevmoo) [inline .gitignore]: Points for sorting these...

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,5 @@
...
+.packages
+.pub/
 packages
```
Comment (kevmoo) [inline .gitignore]: Also: please remove `packages` – not generated in Dart2

**Comment by kevmoo** [comment]
Thanks, @bcko !

**Comment by bcko** [comment]
Thank you for your help @kevmoo ! :)

**Pull request merged**